evm/db/state.py

```diff
@@ -252,169 +320,62 @@ def get_balance(self, address):
             return 0
 
     def set_nonce(self, address, nonce):
-        key = self.nonce_key(address)
-        self._check_accessibility(key)
-
-        self._trie[key] = int_to_big_endian(nonce)
+        validate_canonical_address(address, title="Storage Address")
+        validate_uint256(nonce, title="Nonce")
 
-    def get_nonce(self, address):
-        key = self.nonce_key(address)
+        key = nonce_key(address)
         self._check_accessibility(key)
 
-        if key in self._trie:
-            return big_endian_to_int(self._trie[key])
-        else:
-            return 0
-
-    def set_code(self, address, code):
-        key = self.code_key(address)
-        self._check_accessibility(key)
+        self._trie[key] = int_to_big_endian(nonce)
 
-        self._trie[key] = code
+    def increment_nonce(self, address):
```

Comment:
The implementations of some of these *meta* APIs are the same across these two backends.  Might be suitable to have a `BaseXXXX` backend class whith these common implementations for `increment_nonec, delta_balances,account_has_code_or_nonce,account_is_empty`